modules/juce_opengl/opengl/juce_OpenGLContext.cpp

```diff
@@ -159,7 +159,15 @@ class OpenGLContext::CachedImage  : public CachedComponentImage,
     //==============================================================================
     void paint (Graphics&) override
     {
-        updateViewportSize (false);
+        bool canTriggerUpdate = false;
+#if JUCE_MAC
+        const auto& displays = Desktop::getInstance().getDisplays();
+        context.currentDisplay =
+        &displays.findDisplayForRect (component.getTopLevelComponent()->getScreenBounds());
+        canTriggerUpdate = context.currentDisplay != lastDisplay;
+        lastDisplay = context.currentDisplay;
+#endif
+        updateViewportSize (canTriggerUpdate);

```

Comment:
This could be refactored into `canTriggerUpdateOnPaint()`
And then the call here remains a nice `updateViewportSize (canTriggerUpdateOnPaint());`